framework/src/modules/chain/chain.js

```diff
@@ -253,6 +254,16 @@ module.exports = class Chain {
 					},
 				);
 			}
+
+			// Check if blocks are left in temp_block table
```

Comment:
why don't we do `await this.synchronizor.init()` instead of explicitly calling separate function?

Reply:
I also want to check if there are blocks left in the temp_block table.
This check can possibly be added to the `run` function for block_sync_mechanism?
I prefer this separated logic tho. @shuse2

Reply:
yea, i meant add
`init` function in `synchronizer.js` and check and run there?

Reply:
> yea, i meant add
> `init` function in `synchronizer.js` and check and run there?

This makes sense for me.

Reply:
Does it make sense to still have this `restoreBlocksUponStartup` to separate logic but add a `init` function that calls this `restoreBlocksUponStartup` ?

Reply:
Hmmm yeah II thiink it makes sense to separate the logic within Synchronizer to identify it better